pkg/apis/monitoring/v1/types.go

```diff
@@ -285,6 +285,10 @@ type PrometheusSpec struct {
 	// based on a service monitor can access arbitrary files on the file system
 	// of the Prometheus container e.g. bearer token files.
 	ArbitraryFSAccessThroughSMs ArbitraryFSAccessThroughSMsConfig `json:"arbitraryFSAccessThroughSMs,omitempty"`
+	// OverrideHonorLabels if set to true overrides all user configured honor_labels
+	// if set in ServiceMonitor or PodMonitor to true, this flag will take precedence
+	// and revert the value.
+	OverrideHonorLabels bool `json:"overrideHonorLabels,omitempty"`
```

Comment:
This naming is a bit confusing as it just suggests that `honor_labels` will be overridden but doesn't say with what value.

Reply:
I think with an explaining comment it's ok.